packages/core/src/routes/session.ts

```diff
@@ -27,28 +27,43 @@ export default function sessionRoutes<T extends AnonymousRouter>(router: T, prov
         assertThat(username && password, 'session.insufficient_info');
 
         try {
-          const { id, passwordEncrypted, passwordEncryptionMethod, passwordEncryptionSalt } =
-            await findUserByUsername(username);
+          const {
+            id,
+            passwordEncrypted,
+            passwordEncryptionMethod,
+            passwordEncryptionSalt,
+            accessBlocked,
+          } = await findUserByUsername(username);
 
           assertThat(
             passwordEncrypted && passwordEncryptionMethod && passwordEncryptionSalt,
             'session.invalid_sign_in_method'
           );
+
           assertThat(
             encryptPassword(id, password, passwordEncryptionSalt, passwordEncryptionMethod) ===
               passwordEncrypted,
             'session.invalid_credentials'
           );
 
-          const redirectTo = await provider.interactionResult(
-            ctx.req,
-            ctx.res,
-            {
-              login: { accountId: id },
-            },
-            { mergeWithLastSubmission: false }
-          );
-          ctx.body = { redirectTo };
+          if (accessBlocked) {
+            const accessDeniedError = new RequestError('oidc.access_denied');
+            const redirectUriWithError = await provider.interactionResult(ctx.req, ctx.res, {
```

Comment:
It's better to handle error in middleware.

Reply:
According to [OAuth rfc 4.1.2](https://datatracker.ietf.org/doc/html/rfc6749#section-4.1.2.1), we need to handle this error by redirectUri

> If the resource owner denies the access request or if the request
   fails for reasons other than a missing or invalid redirection URI,
   the authorization server informs the client by adding the following
   parameters to the query component of the redirection URI using the
   "application/x-www-form-urlencoded" format

Reply:
I mean, can we move these code to middleware?